addons/portal/wizard/mail_compose_message.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models
+
+
+class MailComposeMessage(models.TransientModel):
+    _inherit = 'mail.compose.message'
+
+    def default_share_link(self):
+        context = self.env.context
+        active_model = context.get('active_model')
+        active_id = context.get('active_id')
+        model = self.env[active_model]
+        is_model = issubclass(type(model), self.pool['portal.mixin'])
```

Comment:
Why not `isinstance(model, self.pool['portal.mixin'])`?

Or maybe add a relevant hook on e.g. BaseModel doing nothing and override it on portal.mixin? That way people have an easy way of overloading the share link for a specific model, something like that?